api/custom_index.py

```diff
@@ -0,0 +1,215 @@
+                (lane.id, lane.library.name, library.name)
+            )
+        return lane
+
+    def __call__(self, library, annotator, url_for=None):
+        """Render an OPDS navigation feed that lets the patron choose a root
+        lane on their own, without providing any credentials.
+        """
+        if not hasattr(self, 'navigation_feed'):
+            self.navigation_feed = self._navigation_feed(
+                library, annotator, url_for
+            )
+        headers = { "Content-Type": OPDSFeed.NAVIGATION_FEED_TYPE }
+        return Response(unicode(self.navigation_feed), 200, headers)
+
+    def _navigation_feed(self, library, annotator, url_for=None):
+        """Generate an OPDS feed for navigating the COPPA age gate."""
+        url_for = url_for or cdn_url_for
+        base_url = url_for('index', library_short_name=library.short_name)
+
+        # An entry for grown-ups.
+        feed = OPDSFeed(title=library.name, url=base_url)
+        opds = feed.feed
+        yes_url = url_for(
+            'feed',
+            library_short_name=library.short_name,
+            lane_identifier=self.yes_lane_id
+        )
+        opds.append(
+            self.navigation_entry(yes_url, self.YES_TITLE, self.YES_CONTENT)
+        )
+
+        # An entry for children.
+        no_url = url_for(
+            'feed',
+            library_short_name=library.short_name,
+            lane_identifier=self.no_lane_id
+        )
+        opds.append(
+            self.navigation_entry(no_url, self.NO_TITLE, self.NO_CONTENT)
+        )
+
+        # The gate tag is the thing that the SimplyE client actually uses.
+        opds.append(self.gate_tag(self.URI, yes_url, no_url))
+
+        # Add any other links associated with this library, notably
+        # the link to its authentication document.
+        if annotator:
+            annotator.annotate_feed(feed, None)
+        return feed
+
+    @classmethod
+    def navigation_entry(cls, href, title, content):
+        """Create an <entry> that serves as navigation."""
+        E = OPDSFeed.E
+        content_tag = E.content(type="text")
+        content_tag.text = unicode(content)
+        entry = E.entry(
+            E.id(href),
+            E.title(unicode(title)),
+            content_tag,
+        )
+        OPDSFeed.add_link_to_entry(
+            entry, href=href, rel="subsection",
+            type=OPDSFeed.ACQUISITION_FEED_TYPE
+        )
+        return entry
+
+    @classmethod
+    def gate_tag(cls, restriction, met_url, not_met_url):
+        """Create a simplified:gate tag explaining the boolean option
+        the client is faced with.
+        """
+        # The original Instant Classics OPDS feed incorrectly omitted
+        # the simplified: namespace on 'restriction-met' and
+        # 'restriction-not-met', and the SimplyE client depends on the
+        # namespace not being present. For now, we include both a
+        # namespaced and a non-namespaced version.
```

Comment:
I see the namespace on https://instantclassics.librarysimplified.org/index.xml. Is it currently broken in the app?